x-pack/plugins/security_solution/server/lib/detection_engine/signals/bulk_create_threshold_signals.ts

```diff
@@ -73,52 +74,145 @@ const getTransformedHits = (
       logger.warn(`No hits returned, but totalResults >= threshold.value (${threshold.value})`);
       return [];
     }
+    const timestampArray = get(timestampOverride ?? '@timestamp', hit.fields);
+    if (timestampArray == null) {
+      return [];
+    }
+    const timestamp = timestampArray[0];
+    if (typeof timestamp !== 'string') {
+      return [];
+    }
 
     const source = {
-      '@timestamp': get(timestampOverride ?? '@timestamp', hit.fields),
+      '@timestamp': timestamp,
       threshold_result: {
+        terms: [
+          {
+            value: ruleId,
+          },
+        ],
         count: totalResults,
-        value: ruleId,
       },
     };
 
     return [
       {
         _index: inputIndex,
-        _id: calculateThresholdSignalUuid(ruleId, startedAt, threshold.field),
+        _id: calculateThresholdSignalUuid(ruleId, startedAt, threshold.field as string[]),
         _source: source,
       },
     ];
   }
 
-  if (!results.aggregations?.threshold) {
+  const aggParts = results.aggregations && getThresholdAggregationParts(results.aggregations);
+  if (!aggParts) {
     return [];
   }
 
-  return results.aggregations.threshold.buckets
-    .map(
-      ({ key, doc_count: docCount, top_threshold_hits: topHits }: ThresholdAggregationBucket) => {
-        const hit = topHits.hits.hits[0];
-        if (hit == null) {
-          return null;
+  const getCombinations = (buckets: TermAggregationBucket[], i: number, field: string) => {
+    return buckets.reduce((acc: MultiAggBucket[], bucket: TermAggregationBucket) => {
+      if (i < threshold.field.length - 1) {
+        const nextLevelIdx = i + 1;
+        const nextLevelAggParts = getThresholdAggregationParts(bucket, nextLevelIdx);
+        if (nextLevelAggParts == null) {
+          throw new Error('Something went horribly wrong');
         }
-
-        const source = {
-          '@timestamp': get(timestampOverride ?? '@timestamp', hit.fields),
-          threshold_result: {
-            count: docCount,
-            value: key,
-          },
+        const nextLevelPath = `['${nextLevelAggParts.name}']['buckets']`;
+        const nextBuckets = get(nextLevelPath, bucket);
+        const combinations = getCombinations(nextBuckets, nextLevelIdx, nextLevelAggParts.field);
+        combinations.forEach((val) => {
+          const el = {
+            terms: [
+              {
+                field,
+                value: bucket.key,
+              },
+              ...val.terms,
+            ],
+            cardinality: val.cardinality,
+            topThresholdHits: val.topThresholdHits,
+            docCount: val.docCount,
+          };
+          acc.push(el);
+        });
+      } else {
+        const el = {
+          terms: [
+            {
+              field,
+              value: bucket.key,
+            },
+          ],
+          cardinality:
+            threshold.cardinality_field != null
```

Comment:
Rule fails to execute if you leave `cardinality_field` blank in the UI
![image](https://user-images.githubusercontent.com/55718608/108124735-8896c980-7075-11eb-909b-5d7b916f46c4.png)
Simplest change would be to make this check `isEmpty` instead, but ideally we wouldn't allow rules with empty strings for cardinality_field.